### ISSUE TITLE
vcpkg manifest mode dependencies default-features not work after platform filter implemented

### ISSUE THREAD

**Issue by linquize** [microsoft/vcpkg#12693]
After platform filter implemented, it breaks `default-features` again
vcpkg: f225ba822b55a2ba9d3cc2a3c4e0646c8f47f035 #12548
I want to perform `vcpkg install libarchive[core]` in manifest mode.
Note: `libarchive` has a list of `Default-Features`: `bzip2, libxml2, lz4, lzma, lzo, openssl, zstd`

Here is my `vcpkg.json`
```json
{
  "name": "project1",
  "version-string": "0.0.1",
  "dependencies": [
    {
      "name": "libarchive",
      "default-features": false
    },
    "jsoncpp"
  ]
}
```

**Actual** Result:
`libarchive[core]` is not followed, but install `libarchive[bzip2,core,libxml2,lz4,lzma,lzo,openssl,zstd]`
```
The following packages will be built and installed:
  * bzip2[core]:x64-windows
    jsoncpp[core]:x64-windows
    libarchive[bzip2,core,libxml2,lz4,lzma,lzo,openssl,zstd]:x64-windows
  * libiconv[core]:x64-windows
  * liblzma[core]:x64-windows
  * libxml2[core]:x64-windows
  * lz4[core]:x64-windows
  * lzo[core]:x64-windows
  * openssl[core]:x64-windows
  * openssl-windows[core]:x64-windows
  * xxhash[core]:x64-windows
  * zlib[core]:x64-windows
  * zstd[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
```

**Expected** result: Only `libarchive[core]` is installed
```
The following packages will be built and installed:
    jsoncpp[core]:x64-windows
    libarchive[core]:x64-windows
  * zlib[core]:x64-windows
Additional packages (*) will be modified to complete this operation.
```

**Comment by strega-nil** [comment]
Now this is an issue with vcpkg more generally -- #11602 

We need serious discussion on how to solve this.

**Comment by JackBoosY** [comment]
Duplicate with #11602.

**Comment by linquize** [comment]
Fixed in c771e7bd07c3137d43cdca96efcb954439133532
